.github/workflows/bugbot.yaml

```diff
@@ -17,14 +19,151 @@ jobs:
+            const instructionMessage = "reply 'AI code re-review requested' to have bugbot scan again";
+            const instructionExists = sortedComments.some(
+              comment => comment.body.trim() === instructionMessage
+            );
+            console.log(`Instruction message exists: ${instructionExists}`);
+            
+            // Handle initial PR event - post initial comment and instruction
+            if (context.eventName === 'pull_request_target') {
+              console.log('Processing pull_request_target event...');
+              
+              if (!initialBugbotRunExists) {
+                console.log("Posting initial 'bugbot run' comment...");
+                await github.rest.issues.createComment({
+                  issue_number: prNumber,
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  body: 'bugbot run'
+                });
+                console.log("Successfully posted 'bugbot run' comment");
+              } else {
+                console.log("Skipping 'bugbot run' comment - already exists");
+              }
+              
+              // Post instruction message if it doesn't exist
+              if (!instructionExists) {
+                console.log('Posting instruction message...');
+                await github.rest.issues.createComment({
+                  issue_number: prNumber,
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  body: instructionMessage
+                });
+                console.log('Successfully posted instruction message');
+              } else {
+                console.log('Skipping instruction message - already exists');
+              }
+            }
+            
+            // Handle comment event - check for 'AI code re-review requested'
+            if (context.eventName === 'issue_comment') {
+              console.log('Processing issue_comment event...');
+              const commentBody = context.payload.comment.body.trim();
+              console.log(`Processing comment: "${commentBody}"`);
+              
+              // Check if comment contains the re-review request
+              const isReReviewRequest = commentBody === 'AI code re-review requested' || 
+                  commentBody.toLowerCase().includes('ai code re-review requested');
+              console.log(`Is re-review request: ${isReReviewRequest}`);
+              
+              if (isReReviewRequest) {
+                const requestCommentId = context.payload.comment.id;
+                const requestCommentTime = new Date(context.payload.comment.created_at);
+                console.log(`Re-review request comment ID: ${requestCommentId}`);
+                console.log(`Re-review request comment time: ${requestCommentTime.toISOString()}`);
+                
+                // Check if a 'bugbot run' comment was posted after this request
```

Comment:
Can you tell me what you are trying to do here? Can you point me to the trigger which creates comment "bugbot run" when you request for review

Reply:
Are you checking the comment history and verifying if all "request review" has a "bugbot run"? Do we need to do that if we can get the comment body for the very comment that triggered the pipeline?